nautobot/extras/models/datasources.py

```diff
@@ -167,3 +167,56 @@ def sync(self, user, dry_run=False):
         if dry_run:
             return enqueue_git_repository_diff_origin_and_local(self, user)
         return enqueue_pull_git_repository_and_refresh_data(self, user)
+
+    def clone(self, branch=None, head=None):
+        """
+        Perform a shallow clone of the Git repository.
+
+        Args:
+            branch (str): The branch to checkout. If not set, the GitRepository.branch will be used.
+            head (str): Optional Git commit hash to check out instead of pulling branch latest.
+
+        Returns:
+            GitRepository
+        """
+        import copy
+        import uuid
+
+        # import tempfile
+        from nautobot.core.utils.git import GitRepo
+        from nautobot.extras.datasources.git import get_repo_from_url_to_path_and_from_branch
+
+        # tempfile.mkdtemp(dir=self.filesystem_path + "_clone")
```

Comment:
I am confused about where this method call would come in and play a role.

Reply:
Looks like right now you're having it clone into GIT_ROOT, rather than /tmp. Per the comment on #6487, I think we'd decided we wanted the caller to be able to specify which directory to clone to?